wrappers/python/simtk/openmm/app/amberprmtopfile.py

```diff
@@ -248,4 +254,16 @@ def createSystem(self, nonbondedMethod=ff.NoCutoff, nonbondedCutoff=1.0*unit.nan
                 force.setEwaldErrorTolerance(ewaldErrorTolerance)
         if removeCMMotion:
             sys.addForce(mm.CMMotionRemover())
+
+        if switchDistance and nonbondedMethod is not ff.NoCutoff:
+            # make sure it's legal
+            if switchDistance >= nonbondedCutoff:
```

Comment:
This comparison fails if `switchDistance` is a unit and `nonbondedCutoff` is not (or vice versa) (and it fails in a weird way, depending on which is a unit and which is not).  Probably best to convert both to nanometer scalars and compare the numbers that way:

```Python
if unit.is_quantity(switchDistance):
    switchDistance = switchDistance.value_in_unit(unit.nanometer)
if unit.is_quantity(nonbondedCutoff):
    nonbondedCutoff = nonbondedCutoff.value_in_unit(unit.nanometer)
```